src/ai/backend/common/events/user_event/user_event.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+from typing import Any, Mapping, Optional
+
+
+class UserEvent(ABC):
+    @abstractmethod
+    def event_name(self) -> Optional[str]:
+        """Get the name of the event."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def retry_count(self) -> Optional[int]:
+        """Get the retry count for the event."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def user_event_dict(self) -> Mapping[str, Any]:
```

Comment:
If the method name is dict, how about returning dict?

Reply:
Rename to `user_event_mapping`